salina/agents/xformers_transformers.py

```diff
@@ -151,9 +161,13 @@ def forward(self, t: Optional[int] = None, **_):
             keys, values, queries = tokens, tokens, tokens
 
             T = queries.size()[1]
-            attn_mask = (
-                self._get_mask(T, self.n_steps).unsqueeze(0).expand(queries.shape[0] * self.n_heads, -1, -1)
-            ).to(tokens.device)  # (batch * heads) x n_steps x n_steps
+
+            attn_mask = self._get_mask(T, self.n_steps, tokens.device)  # n_steps x n_steps
+
+            if not attn_mask.is_sparse:
```

Comment:
not sure that this is needed actually, I think that we catch that case internally already